src/types.rs

```diff
@@ -42,7 +42,12 @@ pub fn rewrite_path(context: &RewriteContext,
 
     if let Some(ref qself) = qself {
         result.push('<');
-        result.push_str(&pprust::ty_to_string(&qself.ty));
+        let fmt_ty = if let Some(str) = qself.ty.rewrite(context, width, offset) {
```

Comment:
Not at this level. We can just abandon formatting the path entirely. At some point in the call stack we will push the original string, if we can really find no other way to format things.